### PR TITLE
Revise Creation Tree Search Box

### PR DESCRIPTION
**Pull request by oOKexOo**
**When merged this pull request will:**
- Fix the issue that searches are wrong when the side tab is switched.
- Integrates dzn Zeus Search Patch:
  - Searches are now triggered by the enter key.
  - The patch can be disabled via CBA settings.
- Use CBA settings subcategories for faction and module filter
- Fix the issue that multi-side factions like FIA cannot be configured for each side individually.

### CODE REVIEW

File: @AresModAchillesExpansion/addons/language_f/stringtable.xml [hunk 1]
```diff
@@ -6511,6 +6511,13 @@
                 <Chinesesimp>默认摺叠模块层级</Chinesesimp>
                 <Chinese>默認摺疊模塊層級</Chinese>
             </Key>
+            <Key ID="STR_AMAE_ENABLE_DZN_SEARCH_PATCH">
+                <Original>Enable search on enter patch</Original>
+                <English>Enable search on enter patch</English>
+				<Russian>Включить Поиск только по Enter</Russian>
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/language_f/stringtable.xml]: Tabs

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp [hunk 2]
```diff
@@ -1,17 +1,36 @@
+#define TEST "test"
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: ?

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp [hunk 3]
```diff
@@ -1,17 +1,36 @@
...
-		// GUI EDITOR OUTPUT START (by Kex, v1.063, #Lekizy)
-		////////////////////////////////////////////////////////
+		class CreateSearch : RscEdit {};
+		class CreateSearchMaster: CreateSearch
+		{
+			idc = 284;
+			// transfer result of master to slave when enter is pressed
+			onKeyUp = "params ['_ctrlMaster', '_key']; \
+				private _display =  ctrlParent _ctrlMaster; \
+				if (_key in [0x1C, 0x9C]) then \
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: DIK keys?

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp [hunk 4]
```diff
@@ -1,17 +1,36 @@
...
-		
-		////////////////////////////////////////////////////////
-		// GUI EDITOR OUTPUT START (by Kex, v1.063, #Lekizy)
-		////////////////////////////////////////////////////////
+		class CreateSearch : RscEdit {};
+		class CreateSearchMaster: CreateSearch
+		{
+			idc = 284;
+			// transfer result of master to slave when enter is pressed
+			onKeyUp = "params ['_ctrlMaster', '_key']; \
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: Isn't there a better way without using these slashes?

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp [hunk 5]
```diff
@@ -1,17 +1,36 @@
...
+				private _display =  ctrlParent _ctrlMaster; \
+				if (_key in [0x1C, 0x9C]) then \
+				{ \
+					(_display displayCtrl 283) ctrlSetText (ctrlText _ctrlMaster); \
+				}; \
+				true; \
+			";
+			// Makes sure that display key event handlers are disabled when searching
+			onSetfocus = "missionNamespace setVariable ['RscDisplayCurator_search', true];";
+			onKillfocus = "missionNamespace setVariable ['RscDisplayCurator_search', false];";
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: Isn't it onKillFocus?

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp [hunk 6]
```diff
@@ -1,17 +1,36 @@
...
+			onKeyUp = "params ['_ctrlMaster', '_key']; \
+				private _display =  ctrlParent _ctrlMaster; \
+				if (_key in [0x1C, 0x9C]) then \
+				{ \
+					(_display displayCtrl 283) ctrlSetText (ctrlText _ctrlMaster); \
+				}; \
+				true; \
+			";
+			// Makes sure that display key event handlers are disabled when searching
+			onSetfocus = "missionNamespace setVariable ['RscDisplayCurator_search', true];";
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: onSetFocus?

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp (hunk 3 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: Preprocessors are not working in strings. You also can't place them between the strings. Apparently it's not a pure C preprocessor.

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp (hunk 4 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: Nope, it's how you do it in configs

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp (hunk 5 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: I'll fix it, but Arma ignores cases anyway.

File: @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp (hunk 3 above)
Comment (CreepPork) [reply @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: Are comments supported, if so then add a comment which key it is

Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/ui_f/dialogs/Replacement/RscDisplayAttributesModuleTree.hpp]: Nope, but I improved the comment above

**Pull request merged**